### PR TITLE
chore: updating input currency panel to handle comma separator

### PR DESCRIPTION
**Pull request by JackShort**
* updates swap currency input to use commas for decimals based of the locale
* note this is calling a hook that returns the default locale if the `currencyConversion` flag isn't enabled so you have to enable the feature flag to test

### Before
|   Desktop    |
| ------------ | ------------ |
| Screen Shot 2023-09-18 at 3 55 14 PM|

### After
|    Desktop   |
| ------------ | ----------- |
| Screen Shot 2023-09-18 at 3 55 38 PM |

### CODE REVIEW

**Review by JackShort — COMMENTED** [review]
note: when value is actually processed the commas are replaced to periods no matter what

File: src/components/NumericalInput/index.tsx [hunk 1]
```diff
@@ -39,6 +41,13 @@ const StyledInput = styled.input<{ error?: boolean; fontSize?: string; align?: s
   }
 `
 
+function localeUsesComma(locale: SupportedLocale): boolean {
+  const n = 1.1
```
Comment (just-toby) [inline src/components/NumericalInput/index.tsx:45]: no need for a variable here

**Review by just-toby — APPROVED** [review]
nice!

File: src/components/NumericalInput/index.tsx [hunk 2]
```diff
@@ -39,6 +41,13 @@ const StyledInput = styled.input<{ error?: boolean; fontSize?: string; align?: s
...
+  const decimalSeparator = new Intl.NumberFormat(locale).format(n)[1]
```
Comment (zzmp) [inline src/components/NumericalInput/index.tsx:46]: What a great hack. Thanks, ECMAScript.

```suggestion
  const decimalSeparator = new Intl.NumberFormat(locale).format(1.1)[1]
```

**Review by zzmp — APPROVED** [review]
lgtm but please fix the nit

File: src/components/NumericalInput/index.tsx (hunk 2 above)
Comment (justindomingue) [reply src/components/NumericalInput/index.tsx:46]: should we document this hack in code comments?

**Review by just-toby — APPROVED** [review]

**Pull request merged**